operator/pkg/manifest/shared.go

```diff
@@ -537,6 +539,20 @@ func overlaySetFlagValues(iopYAML string, setFlags []string) (string, error) {
 	return string(out), nil
 }
 
+// addDefaultProfileFlagIfNotSet adds the `profile=default` value to set flags if no profile value is set.
+func addDefaultProfileFlagIfNotSet(setFlags []string) []string {
```

Comment:
I don't think this is the right layer to apply this fix. What about something more like
```diff
--- a/operator/pkg/manifest/shared.go
+++ b/operator/pkg/manifest/shared.go
@@ -212,6 +212,9 @@ func GenIOPFromProfile(profileOrPath, fileOverlayYAML string, setFlags []string,
        if ns := GetValueForSetFlag(setFlags, "values.global.istioNamespace"); ns != "" {
                finalIOP.Namespace = ns
        }
+       if finalIOP.Spec.Profile == "" {
+               finalIOP.Spec.Profile = name.DefaultProfileName
+       }
        return util.ToYAMLWithJSONPB(finalIOP), finalIOP, nil
 }
```

Reply:
@howardjohn Thanks! Looks much better.